lib/ramble/ramble/application.py

```diff
@@ -1224,6 +1224,21 @@ def _inputs_and_fetchers(self, workload=None):
                 }
         self._input_fetchers = inputs
 
+    register_phase("check_mirror_support", pipeline="mirror", run_before=["mirror_inputs"])
+
+    def _check_mirror_support(self, workspace, app_inst=None):
+        """Check if the given package manager supports mirroring"""
+        pkgman = self.package_manager
+        if pkgman is not None:
+            # Use the presence of a `mirror_software` phase as an indicator.
+            # This is not very robust, a better approach may be to claim support
+            # from individual package managers.
+            mirror_phases = self.get_pipeline_phases("mirror")
+            if "mirror_software" not in mirror_phases:
+                logger.die(
```

Comment:
Does this break mirroring inputs only? If so, I think we should not die here, and maybe change to only skip trying to mirror software.

(i.e. it's fine if we only mirror inputs during the mirror pipeline)

Reply:
Oh very good catch, initially I thought since this check is itself a phase, an invocation of `--phases mirror-inputs` would just work since this check phase doesn't get selected, but then indeed I forgot the `--include-phase-dependencies` option, which would bring in this check phase if specified.

Updated to not error out in such cases. (The code does seem to be somewhat more convoluted now, so definitely open for better ideas.)